runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java

```diff
@@ -382,6 +386,41 @@ public void onGameTick(GameTick event)
 		{
 			notifier.notify("[" + local.getName() + "] has low oxygen!");
 		}
+
+		if (checkFullSpecEnergy())
+		{
+			notifier.notify("[" + local.getName() + "] has restored spec energy!");;
+		}
+		// having this call here avoids repeating the line inside the checkFullSpecEnergy function
+		lastSpecEnergy = client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT);
+	}
+
+	private boolean checkFullSpecEnergy()
+	{
+		if (config.getSpecEnergyThreshold() == 0) {
+			return false;
+		}
+		if (client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT) != lastSpecEnergy && lastSpecEnergy < (config.getSpecEnergyThreshold() * 10))
```

Comment:
You should just be calling `client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT)` once per tick, not four times.

Reply:
Very true. I made changes to call it once per tick.